File: backend/springProject/src/main/java/io/github/uchkun07/travelsystem/entity/Slideshow.java
package io.github.uchkun07.travelsystem.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 轮播图表实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("slideshow")
public class Slideshow {

    /**
     * 轮播图唯一标识
     */
    @TableId(value = "slideshow_id", type = IdType.AUTO)
    private Integer slideshowId;

    /**
     * 轮播图标题
     */
    @TableField("title")
    private String title;

    /**
     * 副标题/描述
     */
    @TableField("subtitle")
    private String subtitle;

    /**
     * 轮播图图片URL
     */
    @TableField("image_url")
    private String imageUrl;

    /**
     * 关联的景点ID（当link_type为attraction时使用）
     */
    @TableField("attraction_id")
    private Long attractionId;

    /**
     * 显示顺序（数字越小越靠前）
     */
    @TableField("display_order")
    private Integer displayOrder;

    /**
     * 状态：0-禁用，1-启用
     */
    @TableField("status")
    private Integer status;

    /**
     * 开始展示时间
     */
    @TableField("start_time")
    private LocalDateTime startTime;

    /**
     * 结束展示时间
     */
    @TableField("end_time")
    private LocalDateTime endTime;

    /**
     * 点击次数
     */
    @TableField("click_count")
    private Integer clickCount;

    /**
     * 创建时间
     */
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend-apps/homepage/src/apis/slideshow.ts
import { get, post } from "./request";

// 轮播图数据接口
export interface Slideshow {
  slideshowId: number;
  title: string;
  subtitle?: string;
  imageUrl: string;
  attractionId?: number;
  displayOrder: number;
  status: number;
  startTime?: string;
  endTime?: string;
  clickCount: number;
  createTime: string;
  updateTime: string;
}

// API 响应接口
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

/**
 * 获取启用的轮播图(前台展示)
 */
export function getActiveSlideshow() {
  return get<ApiResponse<Slideshow[]>>("/api/home/slideshow/list");
}

/**
 * 记录轮播图点击
 */
export function recordClick(slideshowId: number) {
  return post<ApiResponse<null>>(`/api/home/slideshow/click/${slideshowId}`);
}

/**
 * 获取所有轮播图(管理后台)
 */
export function getAllSlideshow() {
  return get<ApiResponse<Slideshow[]>>("/slideshow/list");
}

/**
 * 根据ID获取轮播图详情
 */
export function getSlideshowById(id: number) {
  return get<ApiResponse<Slideshow>>(`/slideshow/${id}`);
}
